wrangler-service/src/main/java/io/cdap/wrangler/service/directive/WorkspaceHandler.java

```diff
@@ -253,6 +261,25 @@ public void execute(HttpServiceRequest request, HttpServiceResponder responder,
     });
   }
 
+  private List<Row> executeLocally(String namespace, List<String> directives,
+                                   WorkspaceDetail detail) throws DirectiveLoadException, DirectiveParseException {
+    // load the udd
+    composite.reload(namespace);
+    return executeDirectives(namespace, directives, new ArrayList<>(detail.getSample()));
+  }
+
+
+  private List<Row> executeRemotely(String namespace, List<String> directives,
+                                    WorkspaceDetail detail) throws Exception {
+    LOG.info("Executing remotely");
+    RemoteDirectiveRequest directiveRequest = new RemoteDirectiveRequest(directives,
+                                                                         namespace, detail.getSampleAsBytes());
+    RunnableTaskRequest runnableTaskRequest = RunnableTaskRequest.getBuilder(RemoteExecutionTask.class.getName()).
+      withParam(GSON.toJson(directiveRequest)).build();
+    byte[] bytes = getContext().runTask(runnableTaskRequest);
+    return CommonExecutor.convertToRows(bytes);
```

Comment:
how is error returned from `runTask`? If error is returned, `CommonExecutor#convertToRows ` should make sure its being handled

Reply:
An exception is returned on non 200 responses - https://github.com/cdapio/cdap/blob/8452f58fe423624369573d8c001e4df3c102cd36/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java#L75